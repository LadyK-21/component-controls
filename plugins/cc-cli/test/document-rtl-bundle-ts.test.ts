import path from 'path';
import { runTests } from './run-document-tests';

runTests({
  format: 'ts',
  renderer: 'rtl',
  bundle: path.resolve(__dirname, 'bundle/component-controls.js'),
});
