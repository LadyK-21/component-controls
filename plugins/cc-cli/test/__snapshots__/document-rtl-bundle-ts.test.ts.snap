// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rtl bundle ts 1`] = `
"import * as path from 'path';
import { loadStore } from '@component-controls/store';
const { render: reactRender } = require('@component-controls/render/react');
import { render, act } from '@testing-library/react';

const renderErr = (): void => {
  throw new Error('Could not render the story');
};

describe('Components/Header', () => {
  const store = loadStore(
    require(path.resolve(__dirname, './bundle/component-controls.js')),
  );
  const renderFn = store.config.renderFn || reactRender;
  Object.keys(store.docs).forEach(docId => {
    const doc = store.docs[docId];
    if (doc.stories && doc.stories.length) {
      const stories = doc.stories;
      describe(doc.title, () => {
        stories.forEach(storyId => {
          const story = store.stories[storyId];
          it(story.name, async () => {
            let rendered;
            act(() => {
              rendered = renderFn(story, doc);
            });
            if (!rendered) {
              renderErr();
              return;
            }
            const { asFragment } = render(rendered);
            expect(asFragment()).toMatchSnapshot();
          });
        });
      });
    }
  });
});
"
`;
